ecosystem/sep-0008.md

```diff
@@ -88,79 +83,62 @@ Parameters:
 
-## Account Setup
+Depending on whether issuers want to allow the accounts to maintain offers, issuers can leave the accounts in the `AUTHORIZE_TO_MAINTAIN_LIABILITIES_FLAG` state or completely deauthorize the accounts when completing the operation. See [CAP-18] for the fine-grained control of authorization.
 
-Implementing Regulated Assets requires the participating accounts to be “managed” by the issuer. This is achieved by having the issuer be a co-signer on the account, such that a medium threshold operation cannot be submitted without the issuer’s approval.
+Here is an example of the transaction:
 
-In the future, this requirement can be replaced by introducing protocol level [CoSigned assets](https://github.com/stellar/stellar-protocol/issues/146).
+```
+Operation 1: Issuer uses AllowTrust to fully authorize account A, asset X
+Operation 2: Issuer uses AllowTrust to fully authorize account B, asset X
+Operation 3: Payment from A to B
+Operation 4: Issuer uses AllowTrust to set account B, asset X to AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG state
+Operation 5: Issuer uses AllowTrust to set account A, asset X to AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG state
+```
 
 ## Discussion
 
 ### Should my asset be a regulated asset?
 
 Ideally, no. Implementing Regulated Assets should only be used when absolutely necessary, such as for regulatory reasons. It comes with a substantial operational overhead, added complexity, and burden for users. Issuers should only go down this route if it is absolutely required.
-Alternatively, some other available options are to utilize [SEP006](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0006.md) in order to perform KYC on deposit and withdraw, and/or use [AUTHORIZATION_REQUIRED](https://www.stellar.org/developers/guides/issuing-assets.html#requiring-or-revoking-authorization) which allows issuers to authorize specific accounts to hold their issued assets.
 
-### Why doesn’t the approval service submit transactions to the network?
+### Why doesn’t the approval server submit transactions to the network?
 
 - Separation of concerns between signing transactions and submitting them.
 - Transactions might require more signatures from further regulated asset issuers.
+- Wallets can handle the failure from transaction submission based on the type of errors and reconstruct the transaction.
+
+### Should the ruleset simply sit in the approval server?
+
+- Issuers can configure the ruleset with server configs.
+- The server can read the ruleset from a file and we can format the file so that it has a section about ledger state under Horizon.
+
+### Should the approval server make an API call to the issuer's server for rule checking if any?
+
+- We should define the request and the response format in this case.
```

Comment:
💡 These discussion comments appear to be out-of-scope of the SEP, which I think should be focused on the interaction between the client and the service and the responsibilities of each. What goes on behind the scenes or how the service implements the servers responsibilities is unnecessary to constrain.

Reply:
I feel this can be an important part of the process for the approval server to determine whether the transaction is valid. However, I also think your point makes sense. This SEP-8 should only focus on the interaction between the client and the approval server. Do you think it's worth having another SEP for the interaction between the approval server and the issuer's rule checking server if we decide that this is the way to go compared to baking all the rules in the approval server?

Reply:
If the approval server and issuer rule checking are hosted by the same entity, then I don't think that needs to be prescribed as there's no interoperability concerns and the issuer can build whatever they need is relevant to their use case, region, and concerns.

Reply:
> If the approval server and issuer rule checking are hosted by the same entity, then I don't think that needs to be prescribed as there's no interoperability concerns.

It's true that we probably don't have to prescribe if they build everything on their own.

We are trying to figure out the way that issuers might it useful to host the approval server and use it for rule checking. If they find it more useful that the approval server can make an API call, then it's where the complication can come in because we might build a generalized approval server in our mono repo similar to `webauth` and `recoverysigner`. Without knowing the interface, it's hard to add that functionality.

Reply:
Thinking about it more, it might be very difficult to turn every single rule into implementation in the prototype server. I guess the intention of providing a prototype of the approval server might just be using it as an example.

Reply:
https://github.com/stellar/stellar-protocol/pull/750/commits/0fe94d7a79cb6ddf04ae024fcd95e1f4890a4745